src/Servers/HttpSys/src/RequestProcessing/Response.cs

```diff
@@ -622,6 +635,73 @@ private static void FreePinnedHeaders(List<GCHandle> pinnedHeaders)
             }
         }
 
+        internal unsafe void SerializeTrailers(HttpApiTypes.HTTP_DATA_CHUNK[] dataChunks, int currentChunk, List<GCHandle> pins)
+        {
+            Debug.Assert(currentChunk == dataChunks.Length - 1);
+            Debug.Assert(HasTrailers);
+            Trailers.IsReadOnly = true; // Prohibit further modifications.
+            var trailerCount = 0;
+
+            foreach (var trailerPair in Trailers)
+            {
+                trailerCount += trailerPair.Value.Count;
+            }
+
+            var pinnedHeaders = new List<GCHandle>();
+
+            var unknownHeaders = new HttpApiTypes.HTTP_UNKNOWN_HEADER[trailerCount];
+            var gcHandle = GCHandle.Alloc(unknownHeaders, GCHandleType.Pinned);
+            pinnedHeaders.Add(gcHandle);
+            dataChunks[currentChunk].DataChunkType = HttpApiTypes.HTTP_DATA_CHUNK_TYPE.HttpDataChunkTrailers;
+            dataChunks[currentChunk].trailers.trailerCount = (ushort)trailerCount;
+            dataChunks[currentChunk].trailers.pTrailers = gcHandle.AddrOfPinnedObject();
+
+            try
```

Comment:
Do trailers not share the KNOWN_HEADER types?

Reply:
No, All trailers are unknown.